src/control/server/server.go

```diff
@@ -388,10 +422,22 @@ func Start(log *logging.LeveledLogger, cfg *config.Server) error {
 	sysdb.OnLeadershipGained(func(ctx context.Context) error {
 		log.Infof("MS leader running on %s", hostname())
 		mgmtSvc.startJoinLoop(ctx)
+
+		// On gaining leadership, stop forwarding MS events.
+		eventPubSub.Reset()
+		// Handle rank state change notifications.
+		eventPubSub.Subscribe(events.RASTypeRankStateChange, membership)
+
 		return nil
 	})
 	sysdb.OnLeadershipLost(func() error {
 		log.Infof("MS leader no longer running on %s", hostname())
+
+		// On losing leadership, stop handling MS events.
+		eventPubSub.Reset()
+		// Forward events to new MS leader.
+		eventPubSub.Subscribe(events.RASTypeRankStateChange, eventForwarder)
+
 		return nil
 	})
 
```

Comment:
Start is getting to be a really long function. Not necessary in this PR but it would be nice to start thinking of how to break things into helper functions.

Reply:
yep